src/explore-education-statistics-admin/src/pages/release/data/components/DataFileReplacementPlan.tsx

```diff
@@ -73,6 +79,54 @@ const DataFileReplacementPlan = ({
     () => plan?.footnotes.some(footnote => !footnote.valid) ?? false,
     [plan],
   );
+  const isNewReplaceDsvFeatureEnabled = useFeatureFlag(
+    'enableReplacementOfPublicApiDataSets',
+  );
+
+  const hasDataSetVersionPlan = useMemo<boolean>(
+    () =>
+      isNewReplaceDsvFeatureEnabled &&
+      plan?.apiDataSetVersionPlan !== undefined,
+    [plan, isNewReplaceDsvFeatureEnabled],
+  );
+
+  const hasInvalidLocationMapping = useMemo<boolean>(
+    () =>
+      hasDataSetVersionPlan
+        ? !(
+            plan?.apiDataSetVersionPlan?.mappingStatus?.locationsComplete ??
+            false
+          )
+        : false,
+    [plan, hasDataSetVersionPlan],
+  );
+
+  const hasInvalidFilterMapping = useMemo<boolean>(
+    () =>
+      hasDataSetVersionPlan
+        ? !(
+            plan?.apiDataSetVersionPlan?.mappingStatus?.filtersComplete ?? false
```

Comment:
I'm just curious about how `filtersComplete` will be _equal to_ `hasInvalidFilterMapping`, does complete filters mean invalid filter mapping? The same goes for above with locations, perhaps I'm missing some context